src/Simplify.cpp

```diff
@@ -177,37 +175,71 @@ class Simplify : public IRMutator {
                 *max_val = b.second;
                 return true;
             }
+        } else if (const Broadcast *b = e.as<Broadcast>()) {
+            return const_int_bounds(b->value, min_val, max_val);
+        } else if (const Max *max = e.as<Max>()) {
+            int64_t min_a, min_b, max_a, max_b;
+            if (const Min *min = max->a.as<Min>()) {
+                if (const_int_bounds(min->b, &min_a, &max_a) &&
+                    (min_a == max_a) &&
+                    const_int_bounds(max->b, &min_b, &max_b) &&
+                    (min_b == max_b)) {
+                    // Bound of max(min(x, 10), -5) : [-5, 10]
+                    // Bound of max(min(x, 4), 6) : [6, 6]
+                    *min_val = min_b;
+                    *max_val = std::max(min_a, min_b);
```

Comment:
I think this should be std::max(max_a, max_b), and then you don't need the equality checks in the if condition